capstone-backend/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -86,6 +86,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     } catch (Exception e) {
       System.err.println("Error: " + e.getMessage());
     }
+  }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    try {
+
+      String userID = request.getParameter("id");
+      Map<String, Object> user;
+
+      if (userID == null) {
+        System.err.println("Error:\t" + "Null Id");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+
+      Firestore db = getFirestore();
+
+      DocumentSnapshot document = db.collection("users").document(userID).get().get();
+
+      user = document.getData();
+
+      Gson gson = new Gson();
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(user));
+
```

Comment:
nit: I'd remove this line too. Good use of separating whitespace for readability throughout the `try` block though!